Report an activity (Backend)

zPereiraa opened the pull request.

zPereiraa committed:
feat(domain): implemented Edit Enrollment

related issues: #28

zPereiraa committed:
feat(service): implemented Edit Enrollment

zPereiraa committed:
feat(controller): implemented PUT

related issues: #32

AndreGamito1 committed:
feat: added update on participation.java

AndreGamito1 committed:
feat: added Update Participation Method Test

Related: #52

AndreGamito1 committed:
feat: added updateParticipation in ParticipationService.java

Related: #48

AndreGamito1 committed:
feat: added UpdateParticipation Service test

Related: #50

AndreGamito1 committed:
feat: added updateParticipation method in ParticipationController.java

Related: #44

AndreGamito1 committed:
feat: added UpdateParticipationWebServiceIT.groovy

Related: #46

AndreGamito1 committed:
feat: added deleteParticipation in ParticipationService.java

Related: #49

AndreGamito1 committed:
feat: added deleteParticipation method in ParticipationController.java

Related: #45

AndreGamito1 committed:
feat: added deleteParticipation Service test

Related: #51

AndreGamito1 committed:
feat: added deleteParticipation WebService Test

Related: #47

zPereiraa committed:
feat(domain test): implemented UpdateEnrollmentMethodTest

zPereiraa committed:
feat(Service Test): implemented UpdateEnrollmentServiceTest

related issues: #40

zPereiraa committed:
feat(domain): implemented Delete Enrollment and fix Edit Enrollment

related issues: #28 #29

zPereiraa committed:
feat(WebService Test): implemented UpdateEnrollmentWebServiceIT

related issues: #38

AndreGamito1 committed:
feat: added delete method in ParticipationService.java and updated necessary files to support delete operation

Related: #43

AndreGamito1 committed:
feat: added delete participation domain tests

Related: #53

zPereiraa committed:
feat(controller): implemented DELETE

related issues: #33

zPereiraa committed:
feat(domain): fix Delete and Edit

related issues: #28 #29

zPereiraa committed:
feat(Service): implemented Delete Enrollment

related issues: #35

Commit:
feat(domain test): updated UpdateEnrollmentMethodTest

related issues: #36

zPereiraa committed:
feat(domain/serice/webservice tests):

	-implemented DeleteEnrollmentMethodTest
	-implemented DeleteEnrollmentServiceTest
	-updated UpdateEnrollmentServiceTest
	-implemented DeleteEnrollmentWebService

related issues: #37 #39 #40 #41

zPereiraa committed:
feat(): small changes in some files

zPereiraa committed:
feat(fix): small change

zPereiraa committed:
feat(fix): small changes

zPereiraa committed:
feat(fix): small changes

zPereiraa committed:
feat(fix): fix some errors

zPereiraa committed:
feat(fix): fix some erros

zPereiraa committed:
feat(domain/service tests): fix some errors

relating issues: #37 #36 #35 #41

zPereiraa committed:
feat(fix): small changes

AndreGamito1 committed:
fix: prevent unauthorized access in participation controller

ritosilva committed:
fix update and remove participation access control

AndreGamito1 committed:
fix: fixed webservice tests

ritosilva committed:
Merge branch 'gamito' into pereira

zPereiraa committed:
feat(domain/service/controller/tests):
      - changed EnrollmentController.java
      - implemented "ENROLLMENT.MANAGER" in HEPermissionEvaluator.java
      - changed Enrollment.java
      -  some changes in UpdateEnrollmentMethodTest, UpdateEnrollmentServiceTest, DeleteEnrollmentWebServiceIT and UpdateEnrollmentWebServiceIT

realted issues: #32 #33 #28 #29 #40 #36

zPereiraa committed:
feat(domain): small changes

ritosilva committed:
Merge branch 'master' into pereira

zPereiraa committed:
feat(button/view):
                - implemented EditEnrollment Button
                - implemented RemoveEnrollment Button
                - implemented VolunteerEnrollmentsView

related issues: #71 #72 #70

zPereiraa committed:
feat(End-to-End Tests):  implemented Remove/Edit Enrollment End-to-End Tests

realted issues: #74 #75 #76 #77 #78

zPereiraa committed:
feat(fix): small changes

zPereiraa committed:
feat(fix): fix errors
        - changed assement End-to-End tests
        - changes in VolunteerEnrollmetView

zPereiraa committed:
Merge remote-tracking branch 'origin/master' into pereira

ritosilva committed:
small fixes

zPereiraa committed:
Merge remote-tracking branch 'origin/master' into pereira

zPereiraa committed:
feat(domain/service/controller):
	- create the "report" entity
	- implemented condition: only a Volunteer can report
	- implemented constraint: this justification can only be readed by a Administrator
	- implemented condition: report is associated to the volunteer who reported
	- implemented condition: only can be reported until the end of the activity
	- implmented POST: createReport
	- implemented GET: getAdministratorReports

	related issues: #94 #95 #96 #97 #98 #99 #100

zPereiraa committed:
feat(domain/service/controller Tests):
	- Test Domain: CreateReportMethodTest
	- Test Service: CreateReportServiceTest
	- Test WebService: CreateReportWebServiceIT

	related issues: #101 #102 #103

zPereiraa committed:
Merge master into pereira

zPereiraa committed:
feat(fix): small change

zPereiraa committed:
Merge master into pereira

zPereiraa committed:
feat(fix): small change

ritosilva commented on backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/report/domain/Report.java:
```diff
@@ -0,0 +1,103 @@
+    }
+
+    private void justificationIsRequired() {
+        if (this.justification == null || this.justification.trim().length() > 256) {
```
Can you consider a max size?

ritosilva commented on backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/report/ReportController.java:
```diff
@@ -0,0 +1,42 @@
+
+    @GetMapping("/reports/administrator")
+    @PreAuthorize("(hasRole('ROLE_ADMIN'))")
+    public List<ReportDto> getAdministratorReports(Principal principal) {
```
It make no sense, it is getting his own reports, which is none, since the admin cannot report

ritosilva commented on backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/report/ReportService.java:
```diff
@@ -0,0 +1,56 @@
+    public List<ReportDto> getAdministratorReports(Integer userId) {
+        if (userId == null) throw new HEException(USER_NOT_FOUND);
+
+        return reportRepository.getReportsForVolunteerId(userId).stream()
```
These are not administrator reports

ritosilva reviewed (changes requested):
Tests are also missing for get reports

ritosilva committed:
Merge branch 'master' into pereira

zPereiraa committed:
feat(Controller/Service/ Service Tests/WebServices Test): some changes

      -implement GetVolunteerReportsWebServiceIT
      -implement GetVolunteerReportsServicesTest
      -changes in RerportController and ReportService

      related issues: #119 #120 #100

zPereiraa committed:
feat(fix): changes

zPereiraa committed:
feat(fix): changes

zPereiraa committed:
feat(fix): changes

zPereiraa committed:
feat(fix): some changes

ritosilva commented on backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/report/domain/Report.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    private void justificationIsRequired() {
+        if (this.justification == null || this.justification.trim().length() > JUSTIFICAITON_MAX_SIZE) {
```
Justification should be between a min and a max size

ritosilva commented on backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/humanaethica/report/service/GetVolunteerReportsServiceTest.groovy:
```diff
@@ -0,0 +1,99 @@
+
+        when:
+        def reportsVolunteerOne = reportService.getVolunteerReports(volunteerOne.id)
+        def reportsVolunteerTwo = reportService.getVolunteerReports(volunteerTwo.id)
```
Should not test two things at the same time. Get the reports of the activity and check that the result contains the reports of both volunteers

ritosilva commented on backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/report/ReportController.java:
```diff
@@ -0,0 +1,37 @@
+public class ReportController {
+    @Autowired
+    ReportService reportService;
+
```
It is necessary to have a service that given an activity get all its reports

ritosilva reviewed (changes requested):
Some suggestions for change

zPereiraa committed:
feat(fix): some changes

zPereiraa committed:
feat(Controller/Service/Service Test/WebServiceTest):
        -implemented GET: getActivtyReports
        -implemented Service: getReportsByActivity
        -implemented GetReportsByActivityServiceTest
        -implemented GetReportsByActivityWebServiceTest

        related issues: #121 #122 #123

zPereiraa committed:
feat(fix): small change

zPereiraa committed:
feat(fix): fix errors
